### DISCUSSION TITLE
gRPC Schema Registries  [Roadmap Waypoint]

### DISCUSSION THREAD

**Discussion by vordimous** [aklivity/zilla#736] [Ideas]
We are checking in with the Zilla Roadmap to get your thoughts about this new Idea to pull gRPC service definitions from an external schema registry with Zilla #697.

Currently, Zilla can use any `.proto` file that is on the same host where Zilla is running. You can control what services Zilla implements by controlling the files the runtime can access. Any file management tooling will work with Zilla to update the proto service definitions.

We want to include more ways to manage gRPC services on top of Kafka and have been working hard to include new general support for external schema registries. With the #697 roadmap story, the goal is to enable more flexibility when defining and updating these services.

Check out the feature spec, ask questions, and tell us what you think might improve it!

[Join the discussion](https://github.com/aklivity/zilla/discussions) to make Zilla even better! Star :star: the [Zilla repo](https://github.com/aklivity/zilla) to watch for new updates and come hang out in our [Slack](https://www.aklivity.io/slack).

**Comment by dudo** [comment]
I like the idea of keeping the source of truth within the schema registry. It also opens up the possibility of adding magic bytes to the protobuf payloads, which is necessary to even use protobuf for most connector converters.

**Reply by vordimous** [reply]
Good point! Protobufs aren't worth as much when devs just see raw bytes, even if everything is working.
